nelox/interpreter.py

```diff
@@ -0,0 +1,158 @@
+    env.define("/", lambda *args: _apply(operator.truediv, *args))
+    env.define(">", _comparison(operator.gt))
+    env.define("<", _comparison(operator.lt))
+    env.define(">=", _comparison(operator.ge))
+    env.define("<=", _comparison(operator.le))
+    env.define("=", _comparison(operator.eq))
+    env.define("print", lambda *args: print(*args))
+
+
+class Interpreter:
+    def __init__(self):
+        self.global_env = Environment()
+        _define_builtins(self.global_env)
+
+    def interpret(self, expressions):
+        result = None
+        for expr in expressions:
+            result = self.evaluate(expr, self.global_env)
+            if "_" in self.global_env.values:
+                self.global_env.set("_", result)
+            else:
+                self.global_env.define("_", result)
+        return result
+
+    def evaluate(self, expr, env):
+        if isinstance(expr, Literal):
+            return expr.value
+
+        elif isinstance(expr, Variable):
+            return env.get(expr.name.lexeme)
+
+        elif isinstance(expr, List):
+            if not expr.elements:
+                return None
+
+            head = expr.elements[0]
+            args = expr.elements[1:]
+
+            if isinstance(head, Variable):
+                name = head.name.lexeme
+
+                if name == "func":
+                    fn_name = args[0]
+                    lambda_expr = List([
+                        Variable(Token(TokenType.IDENTIFIER, "lambda", None, 0)),
+                        args[1],
+                        args[2]
+                    ])
+                    define_expr = List([
+                        Variable(Token(TokenType.IDENTIFIER, "define", None, 0)),
+                        fn_name,
+                        lambda_expr
+                    ])
+                    return self.evaluate(define_expr, env)
+
+                elif name == "define":
+                    var_name = args[0].name.lexeme
+                    value = self.evaluate(args[1], env)
+                    env.define(var_name, value)
+                    return value
+
+                elif name == "set":
+                    var_name = args[0].name.lexeme
+                    value = self.evaluate(args[1], env)
+                    env.set(var_name, value)
+                    return value
+
+                elif name == "if":
+                    condition = self.evaluate(args[0], env)
+                    branch = args[1] if condition else args[2]
+                    return self.evaluate(branch, env)
+
+                elif name == "lambda":
+                    param_tokens = args[0].elements
+                    if len(args) > 2:
+                        body = List([
+                                        Variable(Token(TokenType.IDENTIFIER, "begin", None, 0))
+                                    ] + args[1:])
+                    else:
+                        body = args[1]
```

Comment:
A simpler solution, I think, is just to use interpret instead of evaluate below (using the list `args[1:]`).